common/src/main/java/com/banica/common/PropertyModifier.java

```
@@ -0,0 +1,52 @@
+package com.banica.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.BiConsumer;
+
+public abstract class PropertyModifier {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PropertyModifier.class);
+
+    protected void modifyPropertySource(Map<String, String> propertyMap, Properties properties,
+                                      BiConsumer<String, String> modifyingFunction, File file) {
+        if (!file.exists()) {
+            try {
+                Files.createDirectories(Paths.get(file.getParent()));
+                Files.createFile(Paths.get(file.getPath()));
+            } catch (IOException e) {
+                e.printStackTrace();
+                LOGGER.warn("IOException caught while trying to create properties file: " + file.getPath());
+                throw new IllegalStateException();
+            }
+        }
+
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+            properties.load(fileInputStream);
+            System.out.println(properties);
+        } catch (IOException e) {
+            LOGGER.warn("IOException caught while trying to load properties file: " + file.getPath());
+            e.printStackTrace();
+            return;
```

It might be better to throw an exception here as well (so that JMX is aware). Also, in the catch below.